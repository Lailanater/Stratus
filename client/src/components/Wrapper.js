import React from 'react';
import { useSelector } from 'react-redux';
import { ThemeProvider } from '@material-ui/styles';
import { SnackbarProvider } from 'notistack';
import createMuiTheme from '@material-ui/core/styles/createMuiTheme';

const Wrapper = props => {
  const theme = useSelector(state => state.theme);
  let currentTheme = createMuiTheme({
    palette: {
      type: theme,
      primary: { main: '#00838F' },
      secondary: { main: '#ed407a' }
    }
  });

  return (
    <ThemeProvider theme={currentTheme}>
      <SnackbarProvider maxSnack={2}>{props.children}</SnackbarProvider>
    </ThemeProvider>
  );
};

export default Wrapper;
