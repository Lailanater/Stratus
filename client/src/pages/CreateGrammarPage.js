import React, { useState } from 'react';
import Typography from "@material-ui/core/Typography";
import TextField from "@material-ui/core/TextField";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Checkbox from "@material-ui/core/Checkbox";
import FormGroup from "@material-ui/core/FormGroup";
import FormControl from "@material-ui/core/FormControl";
import { withSnackbar } from "notistack";
import StepForm from "../components/StepForm";
import OptionPicker from "../components/OptionPicker";
import API from "../routes/API";
import {useSelector} from "react-redux";

const CreateGrammarPage = (props) => {

    const menuName = useSelector(state => state.menuName);
    const needsRepeat = useSelector(state => state.needsRepeat);
    const projectPath = useSelector(state => state.currentProject.projectPath);
    const dtmfOptions = useSelector(state => state.dtmfOptions);
    const [needsDTMF, setNeedsDTMF] = useState(false);
    const [needsVoice, setNeedsVoice] = useState(false);

    const grammarName = (
        <TextField
            label="Menu Name"
            helperText="Enter the name of the menu you are creating grammar(s) for"
            margin="normal"
            variant="outlined"
        />
    );

    const selectGrammarType = (
        <FormControl component="fieldset">
            <Typography>Which Type(s) of Grammars Do You Need?</Typography>
            <FormGroup>
                <FormControlLabel
                    control={<Checkbox checked={needsDTMF} onChange={handleChange} value="dtmf" />}
                    label="DTMF"
                />
                <FormControlLabel
                    control={<Checkbox checked={needsVoice} onChange={handleChange} value="voice" />}
                    label="Voice"
                />
            </FormGroup>
        </FormControl>
    );

    let steps = [];
    addStep("Enter The Name of The Menu to Make Grammar(s) for", grammarName);
    addStep("Select DTMF Options", <OptionPicker maxOptions={9} />);
    addStep("Select Type of Grammar(s)", selectGrammarType);

    function createGrammar() {
        API.createGrammar(menuName, "dtmf", projectPath, needsRepeat, dtmfOptions);
        props.enqueueSnackbar("Grammar was successfully created!", {variant: "success", autoHideDuration: 2000});
    }

    function handleChange(event) {
        if (event.currentTarget.value === "dtmf") {
            setNeedsDTMF(!needsDTMF);
        } else {
            setNeedsVoice(!needsVoice);
        }
    }

    function addStep(label, component) {
        steps = steps.concat(createStep(label, component));
    }

    function createStep(label, component) {
        return {
            label,
            component
        };
    }

    return (
        <StepForm steps={steps} onSubmit={createGrammar} />
    );
};

export default withSnackbar(CreateGrammarPage);