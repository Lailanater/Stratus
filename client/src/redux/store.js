import { createStore } from 'redux';
import rootReducer from './reducers';
import { loadState } from '../localStorage';

let initialState = loadState();

if (initialState === undefined) {
  initialState = {
    isAppMenuDisplayed: false,
    theme: 'dark',
    currentProject: '',
    projects: [],
    menuName: '',
    needsRepeat: false,
    needsDtmf: false,
    needsVoice: false,
    dtmfOptions: [],
    defaultRouteTo: 'menu',
    isNextButtonDisabled: false,
    isFinishButtonDisabled: false
  };
}

export default createStore(rootReducer, initialState);
