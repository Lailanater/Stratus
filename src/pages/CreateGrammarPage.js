import React, {Component} from 'react';
import Typography from "@material-ui/core/Typography";
import Slider from "@material-ui/core/Slider";
import Stepper from "@material-ui/core/Stepper";
import Step from "@material-ui/core/Step";
import StepLabel from "@material-ui/core/StepLabel";
import Paper from "@material-ui/core/Paper";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Checkbox from "@material-ui/core/Checkbox";
import FormGroup from "@material-ui/core/FormGroup";
import FormControl from "@material-ui/core/FormControl";
import {withSnackbar} from "notistack";
import {Link} from "react-router-dom";

class CreateGrammarPage extends Component {
    constructor(props) {
        super(props);

        this.state = {
            activeStep: 0,
            needsDTMF: false,
            needsVoice: false
        };

        this.marks = [
            {
                value: 0,
                label: '0',
            },
            {
                value: 1,
                label: '1',
            },
            {
                value: 2,
                label: '2',
            },
            {
                value: 3,
                label: '3',
            },
            {
                value: 4,
                label: '4',
            },
            {
                value: 5,
                label: '5',
            },
            {
                value: 6,
                label: '6',
            },
            {
                value: 7,
                label: '7',
            },
            {
                value: 8,
                label: '8',
            },
            {
                value: 9,
                label: '9',
            },
        ];

        this.getStepContent = this.getStepContent.bind(this);
        this.goBack = this.goBack.bind(this);
        this.goNext = this.goNext.bind(this);
        this.createGrammar = this.createGrammar.bind(this);
        this.selectGrammarType = this.selectGrammarType.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }


    getStepContent(activeStep) {
        switch (activeStep) {
            case 0: {
                return this.getGrammarName();
            }
            case 1: {
                return this.selectDtmfOptions();
            }
            case 2: {
                return this.selectGrammarType();
            }
            default:
                return this.getGrammarName();
        }
    }

    getGrammarName() {
        return (
            <TextField
                label="Menu Name"
                helperText="Enter the name of the menu you are creating grammar(s) for"
                margin="normal"
                variant="outlined"
            />
        );
    }

    selectDtmfOptions() {
        return (
            <div>
                <Typography id="discrete-slider-custom" gutterBottom>
                    How many options are you going to need?
                </Typography>
                <Slider
                    defaultValue={0}
                    step={1}
                    valueLabelDisplay="auto"
                    max={9}
                    marks={this.marks}
                />
            </div>
        );
    }

    goBack() {
        this.setState({
            activeStep: this.state.activeStep - 1
        });
    }

    goNext() {
        this.setState({
            activeStep: this.state.activeStep + 1
        }, () => {
            console.log(this.state);
        });
    }

    createGrammar() {
        this.props.enqueueSnackbar("Grammar was successfully created!", {variant: "success", autoHideDuration: 2000});
    }

    handleChange(event) {
        if (event.currentTarget.value === "dtmf") {
            this.setState({
                needsDTMF: !this.state.needsDTMF
            });
        } else {
            this.setState({
                needsVoice: !this.state.needsVoice
            });
        }
    }

    selectGrammarType() {
        return (
            <FormControl component="fieldset">
                <Typography>Which Type(s) of Grammars Do You Need?</Typography>
                <FormGroup>
                    <FormControlLabel
                        control={<Checkbox checked={this.state.needsDTMF} onChange={this.handleChange} value="dtmf"/>}
                        label="DTMF"
                    />
                    <FormControlLabel
                        control={<Checkbox checked={this.state.needsVoice} onChange={this.handleChange} value="voice"/>}
                        label="Voice"
                    />
                </FormGroup>
            </FormControl>
        );
    }

    render() {
        return (
            <div>
                <Stepper activeStep={this.state.activeStep}>
                    <Step>
                        <StepLabel>Enter The Name of The Menu to Make Grammar(s) for</StepLabel>
                    </Step>
                    <Step>
                        <StepLabel>Select DTMF Options</StepLabel>
                    </Step>
                    <Step>
                        <StepLabel>Select Type of Grammar(s)</StepLabel>
                    </Step>
                </Stepper>
                <Paper>
                    {this.getStepContent(this.state.activeStep)}
                    <br/>
                    <Button variant="contained" onClick={this.goBack} disabled={this.state.activeStep === 0}>
                        Previous
                    </Button>
                    {this.state.activeStep === 2 ?
                        <Link to="/">
                            <Button variant="contained" color="primary" onClick={this.createGrammar}>
                                Finish
                            </Button>
                        </Link>
                        :
                        <Button variant="contained" color="secondary" onClick={this.goNext}>
                            Next
                        </Button>
                    }
                </Paper>
            </div>
        );
    }
}

export default withSnackbar(CreateGrammarPage);