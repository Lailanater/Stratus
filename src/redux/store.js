import {createStore} from "redux";
import rootReducer from "./reducers";

export const initialState = {
    isAppMenuDisplayed: false,
    theme: "dark",
    currentProject: "",
    projects: []
};

export default createStore(rootReducer);