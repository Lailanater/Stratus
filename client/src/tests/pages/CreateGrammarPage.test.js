import React from 'react';
import renderer from 'react-test-renderer';
import CreateGrammarPage from '../../pages/CreateGrammarPage';
import store from '../../redux/store';
import { Provider } from 'react-redux';
import Wrapper from '../../components/Wrapper';

it('renders without crashing', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Wrapper>
        <CreateGrammarPage />
      </Wrapper>
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
