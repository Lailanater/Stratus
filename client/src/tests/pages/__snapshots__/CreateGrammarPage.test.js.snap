// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div>
    <div
      className="MuiPaper-root MuiPaper-elevation0 MuiStepper-root MuiStepper-horizontal"
    >
      <div
        className="MuiStep-root MuiStep-horizontal"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal"
        >
          <span
            className="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                className="MuiStepIcon-text"
                textAnchor="middle"
                x="12"
                y="16"
              >
                1
              </text>
            </svg>
          </span>
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
            >
              Enter The Name of The Menu to Make Grammar(s) for
            </span>
          </span>
        </span>
      </div>
      <div
        className="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          className="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        className="MuiStep-root MuiStep-horizontal"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
        >
          <span
            className="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiStepIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                className="MuiStepIcon-text"
                textAnchor="middle"
                x="12"
                y="16"
              >
                2
              </text>
            </svg>
          </span>
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              Select DTMF Options
            </span>
          </span>
        </span>
      </div>
      <div
        className="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          className="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        className="MuiStep-root MuiStep-horizontal"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
        >
          <span
            className="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiStepIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                className="MuiStepIcon-text"
                textAnchor="middle"
                x="12"
                y="16"
              >
                3
              </text>
            </svg>
          </span>
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              Repeat?
            </span>
          </span>
        </span>
      </div>
      <div
        className="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          className="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        className="MuiStep-root MuiStep-horizontal"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
        >
          <span
            className="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiStepIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                className="MuiStepIcon-text"
                textAnchor="middle"
                x="12"
                y="16"
              >
                4
              </text>
            </svg>
          </span>
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              Select Type of Grammar(s)
            </span>
          </span>
        </span>
      </div>
    </div>
    <div
      className="container"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Previous
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "marginLeft": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Next
        </span>
      </button>
      <br />
      <div
        className="sc-bwzfXH jQNUeE"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
          >
            Menu Name
            <span
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
              â€‰
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-210 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-211"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-contained Mui-required"
          >
            Enter the name of the menu you are creating grammar(s) for
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
